client/helpers/queued-jobs.ts

```diff
@@ -0,0 +1,101 @@
+import { AxiosResponse } from 'axios';
+import { Failure, Polling, QueuedJob } from '../../index';
+import { isFailure, isQueuedJob, prettyJson } from '../utils';
+
+export const PollIntervalMs = 5000;
+
+export const AttemptsPerMin = 60000 / PollIntervalMs;
+
+export const MaxAttempts = 60 * AttemptsPerMin; // Try for an hour
+
+/** Polling async queued job request arguments. */
+export interface PollQueuedJobReq {
+  /** Queued job ID. */
+  readonly id: string;
+
+  /** Function called to get queued job. */
+  readonly getQueuedJob: (id: string) => Promise<AxiosResponse<QueuedJob>>;
+
+  /** If `true`, doesn't fail if API returns 404 status code. */
+  readonly allow404?: boolean;
+
+  /** {@link Polling} */
+  readonly polling: Polling;
+}
+
+export interface PollQueuedJobRes<T> extends PollJobRes<T> {
+  attempts: number;
+  id: string;
+}
+
+type PollRes<T> = Failure | QueuedJob | T;
+
+interface PollJobRes<T> {
+  res: PollRes<T>;
+  status: number;
+}
+
+/**
+ * Poll `getQueuedJob` until redirected to resulting resource, `error`, or reach
+ * `polling.maxAttempts`.
+ *
+ * @param req - {@link PollQueuedJobReq}.
+ * @returns {@link PollQueuedJobRes}.
+ */
+export async function pollQueuedJob<T extends { data: { id: string } }>({
```

Comment:
Pulled this out of `utils` and instead of throwing, return the error to the caller so it can decide what to do with it.